src/components/InternalOverlay/index.d.ts

```diff
@@ -0,0 +1,17 @@
+import { RefObject, ComponentClass, FunctionComponent } from 'react';
+
+type TriggerElementRefFunction = () => RefObject;
+
+interface Position {
+    top: number;
+    left: number;
+}
+
+export interface InternalOverlayProps {
+    render: ComponentClass | FunctionComponent;
+    isVisible?: boolean;
+    triggerElementRef: RefObject | TriggerElementRefFunction;
+    positionResolver: () => Position;
```

Comment:
this function is passed by parameters an object with this structure

{
        trigger: triggerMeta,
        viewport: viewportMeta,
        content: contentMeta,
    }